arelle/FileSource.py

```diff
@@ -490,49 +586,65 @@ def file(self, filepath, binary=False, stripDeclaration=False, encoding=None):
             elif archiveFileSource.isInstalledTaxonomyPackage:
                 # remove TAXONOMY_PACKAGE_FILE_NAME from file path
                 if filepath.startswith(archiveFileSource.basefile):
+                    assert isinstance(archiveFileSource.basefile, list)
```

Comment:
Wrong but required, otherwise we get this: `arelle/FileSource.py:422: error: Argument 1 to "len" has incompatible type "Union[str, List[str], None]"; expected "Sized"  [arg-type]`

Reply:
I think this is a bug. Both `str` and `list` implement `Sized` and based on the line directly above this it would appear to be a `str`. Use asset not None to be safe.